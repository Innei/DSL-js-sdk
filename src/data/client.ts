import createClient from 'openapi-fetch'
import { components, paths, operations } from '../types/data.js'
import { ClientOptions } from '../types/utils.js'
import { DEFAULT_RSS3_NET } from '../constants.js'
import { Res, debug, fetchWithLog, querySerializer } from '../utils.js'

export type Activity = components['schemas']['Activity']
export type Profile = components['schemas']['Profile']
export type Tag = components['schemas']['Tag']
export type Type = components['schemas']['Type']
export type Direction = components['schemas']['Direction']
export type Platform = components['schemas']['Platform']
export type Network = components['schemas']['Network']
export type TotalPage = components['schemas']['MetaTotalPages']
export type Cursor = components['schemas']['MetaCursor']
export type Media = components['schemas']['Media']
export type Error = components['schemas']['Error']

/**
 * Data client for interacting with the data server.
 */
export function client(opt: ClientOptions = {}) {
  if (!opt.baseUrl) opt.baseUrl = DEFAULT_RSS3_NET + '/data'
  if (!opt.querySerializer) opt.querySerializer = querySerializer

  opt.fetch = fetchWithLog(debug.extend('data').extend('fetch'), opt.fetch)

  const client = createClient<paths>(opt)

  /**
   * Query transactions.
   */
  async function activity(
    id: string,
    query?: operations['GetActivity']['parameters']['query'],
  ): Res<Activity, TotalPage> {
    const { data, error } = await client.GET('/activities/{id}', { params: { path: { id }, query } })
    if (error || !data) throw error

    if (!data.data) return data as never

    if (!data.meta) return { data: data.data }

    return {
      data: data.data,
      meta: data.meta,
      nextPage: () => {
        if (!query) query = {}
        return activity(id, { ...query, action_page: (query.action_page || 0) + 1 })
      },
    }
  }

  /**
   * Query activities.
   */
  async function activities(
    account: string,
    query?: operations['GetAccountActivities']['parameters']['query'],
  ): Res<Activity[], Cursor> {
    const { data, error } = await client.GET('/accounts/{account}/activities', {
      params: {
        path: { account },
        query,
      },
    })
    if (error || !data) throw error

    const list = data.data?.filter((a): a is Activity => !!a) || []

    if (!data.meta) return { data: list }

    return {
      data: list,
      meta: data.meta,
      nextPage: () => {
        if (!data.meta) return {} as never
        return activities(account, { ...query, cursor: data.meta.cursor })
      },
    }
  }

  /**
   * Query activities by multiple accounts.
   */
  async function activitiesBatch(query: components['schemas']['AccountsActivitiesRequest']): Res<Activity[], Cursor> {
    const { data, error } = await client.POST('/accounts/activities', {
      body: query,
    })
    if (error || !data) throw error

    const list = data.data?.filter((a): a is Activity => !!a) || []

    if (!data.meta) return { data: list }

    return {
      data: list,
      meta: data.meta,
      nextPage: () => {
        if (!data.meta) return {} as never
        return activitiesBatch({ ...query, cursor: data.meta.cursor })
      },
    }
  }

  /**
   * Query mastodon activities.
   */
  async function mastodonActivities(
    account: string,
    query?: operations['GetMastodonActivities']['parameters']['query'],
  ): Res<Activity[], Cursor> {
    const client = createClient<paths>(opt)

    const { data, error } = await client.GET('/mastodon/{account}/activities', {
      params: {
        path: { account },
        query,
      },
    })
    if (error || !data) throw error

    if (!data.meta) return data as never

    const list = data.data?.filter((a): a is Activity => !!a) || []

    return {
      data: list,
      meta: data.meta,
    }
  }

  /**
   * Query profiles.
   */
  async function profiles(
    account: string,
    query?: operations['GetAccountProfiles']['parameters']['query'],
  ): Res<Profile[], null> {
    const { data, error } = await client.GET('/accounts/{account}/profiles', {
      params: {
        path: { account },
        query,
      },
    })
    if (error || !data) throw error

    const list = data.data?.filter((a): a is Profile => !!a) || []

    return {
      data: list,
      meta: null,
    }
  }

  /**
   * Query mastodon profiles.
   */
  async function mastodonProfiles(account: string): Res<Profile[], null> {
    const [handle, domain] = account.split('@')

    const data = await fetch(`https://${domain}/api/v2/search?q=${handle}&resolve=false&limit=1`)
      .then((res) => res.json())
      .catch(() => {
        return { data: [] }
      })

    if (data.accounts.length === 0) {
      return { data: [], meta: null }
    } else {
      const profile = data.accounts[0]
      return {
        data: [
          {
            address: `${profile.username}@${domain}`,
            bio: profile.note,
            handle: `${profile.username}@${domain}`,
            name: profile.username,
            network: 'Mastodon',
            platform: 'Mastodon',
            profileURI: [profile.avatar],
            url: profile.url,
          },
        ],
        meta: null,
      }
    }
  }

  /**
   * Query platform activities.
   */
  async function platformActivities(
    platform: components['schemas']['Platform'],
    query?: operations['GetPlatformActivities']['parameters']['query'],
  ): Res<Activity[], Cursor> {
    const { data, error } = await client.GET('/platforms/{platform}/activities', {
      params: {
        path: { platform },
        query,
      },
    })
    if (error || !data) throw error

    const list = data.data?.filter((a): a is Activity => !!a) || []

    if (!data.meta) return { data: list }

    return {
      data: list,
      meta: data.meta,
    }
  }

  return {
    activity,
    activities,
    activitiesBatch,
    mastodonActivities,
    profiles,
    mastodonProfiles,
    platformActivities,
  }
}
