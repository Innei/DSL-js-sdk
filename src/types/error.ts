import { components } from './data'

export type Error = components['schemas']['Error']
