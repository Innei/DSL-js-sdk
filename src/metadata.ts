import { components } from './types/data'
import { metadataDoc } from './metadata-doc'

type NotUndefined<T> = T extends undefined ? never : T

type MakeKeysOptional<T> = {
  [K in keyof T]?: T[K]
}

type Doc = typeof metadataDoc

type Map = MakeKeysOptional<Doc>

function getTagType(action: components['schemas']['Action']): keyof Map {
  return `${action.tag}-${action.type}` as keyof Map
}

type Handlers = {
  [key in keyof Map]: (metadata: NotUndefined<NotUndefined<Map[key]>['ref']>) => void
}

export function handleMetadata(action: components['schemas']['Action'], hs: Handlers) {
  const h = hs[getTagType(action)]
  if (!h) return

  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  h(action.metadata as any)
}
