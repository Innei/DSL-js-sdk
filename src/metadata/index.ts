import { components } from '../types/data'
import { getTagType } from '../utils'
import { metadataDoc } from './doc'

type NotUndefined<T> = T extends undefined ? never : T

type MakeKeysOptional<T> = {
  [K in keyof T]?: T[K]
}

type Doc = typeof metadataDoc

export type TagTypeMap = MakeKeysOptional<Doc>

type Handlers = {
  [key in keyof TagTypeMap]: (metadata: NotUndefined<NotUndefined<TagTypeMap[key]>['ref']>) => void
}

export function handleMetadata(action: components['schemas']['Action'], hs: Handlers) {
  const h = hs[getTagType(action)]
  if (!h) return

  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  h(action.metadata as any)
}
