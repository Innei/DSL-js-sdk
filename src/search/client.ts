import createClient from 'openapi-fetch'
import { paths } from '../types/search'
import { ClientOptions } from '../types/utils'
import { DEFAULT_SEARCH_SERVER } from '../constants'

/**
 * Search client for interacting with the search server.
 */
export function client(opt: ClientOptions = {}) {
  if (!opt.baseUrl) opt.baseUrl = DEFAULT_SEARCH_SERVER

  const client = createClient<paths>(opt)

  return {
    /**
     * Query activities.
     */
    async activities(query: paths['/api/Feed/v2/search']['post']['requestBody']['content']['application/json']) {
      if (!query.platform) query.platform = ['ALL']
      if (!query.network) query.network = ['ALL']
      if (!query.sort) query.sort = 'NONE'
      if (!query.between) query.between = { lte: -1, gte: -1 }

      const { data, error } = await client.post('/api/Feed/v2/search', {
        body: query,
      })
      if (error || !data) throw error

      return data
    },
  }
}
