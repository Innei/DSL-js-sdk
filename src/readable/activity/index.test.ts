import { describe, it } from 'vitest'
import { client } from '../../data/client'
import { metadataDoc } from '../../metadata/doc'
import { formatPlain } from './index'
import { components } from '../../types/data'

export type MetadataDoc = {
  [key: string]: {
    ref: components['schemas']['Metadata']
    tag: components['schemas']['Tag']
    type: components['schemas']['Type']
    actions?: string[]
    platforms?: string[]
    examples?: {
      actionType?: string
      id: string
      comment?: string
    }[]
  }
}

describe('formatPlain should work as expected', () => {
  const c = client()

  Object.entries(metadataDoc as MetadataDoc).map(([key, value]) => {
    value.examples?.forEach((example) => {
      it.concurrent(`${key} ${example.id}`, async ({ expect }) => {
        const { data: activity } = await c.activity(example.id)
        expect(formatPlain(activity)).matchSnapshot()
      })
    })
  })
})
