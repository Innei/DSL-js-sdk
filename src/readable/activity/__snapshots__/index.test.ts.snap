// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`formatPlain should work as expected > collectible-approval 0x000005ec834f55bcd378f2f588e051501af336c1c6259dac20c8a3ee0551f661 1`] = `"Approved \\"SmallBrosNFT Official collection\\" to 0x1E0049783F008A0085193E00003D00cd54003c71"`;

exports[`formatPlain should work as expected > collectible-approval 0x00002f42ad98c0a0960847bcdba90054e8d21c49cadb074fcce4c0a730fff7cd 1`] = `"Revoked the approval of \\"CryptoChads collection\\" to 0x1E0049783F008A0085193E00003D00cd54003c71"`;

exports[`formatPlain should work as expected > collectible-auction 0x0006c67b092640082e392c4a86e2a8cbdcf47a8cc31f9ca31f9f7c87a5cb81f8 1`] = `"Won an auction for \\"GeneraTive and GliTch\\" on \\"Foundation\\""`;

exports[`formatPlain should work as expected > collectible-auction 0x00081a2fcf00d00d644b7998330ef4a795419be906b8b2d575bda475e7ea2ad4 1`] = `"Made a bid for \\"Act of Killing\\" on \\"Foundation\\""`;

exports[`formatPlain should work as expected > collectible-auction 0x000590259fdd6ad57f4071878383ac3c378a88c491409e00a444275bc0d3179b 1`] = `"Created an auction for \\"My Life and My Times\\" on \\"Foundation\\""`;

exports[`formatPlain should work as expected > collectible-auction 0x001a0608a2e4273e441e28a5a3f1d41cc3596da3d8d38c842c5d53a76c374b95 1`] = `"Updated an auction for \\"FORA - FOLLOW MY ART\\" on \\"Foundation\\""`;

exports[`formatPlain should work as expected > collectible-auction 0x0012823ee026f352158ae0b1ceaaef3f7ac7b757fc49501717fc6363455a601b 1`] = `"Canceled an auction for \\"About me \\" on \\"Foundation\\""`;

exports[`formatPlain should work as expected > collectible-auction 0x2787fe62e3f31207524217fed8788329560da5964dbf90f791651aaac00b9131 1`] = `"Bought \\"UNRIVALLED\\" from 0x9b0Da6499d6eA156c2FBCa0f2104d42B7f8f2F88 on \\"Foundation\\""`;

exports[`formatPlain should work as expected > collectible-burn 0x000071563c854e34c2534e7e961a53dd648411b62467d6c10b38a80b51d1f51e 1`] = `"Burned \\"THE MAZE by TAYLOR\\""`;

exports[`formatPlain should work as expected > collectible-mint 0x000002803fd033274860a06bb8b7285c0595f550b73e621a92b7095fcd79afaa 1`] = `"Minted \\"Chibi Valkyrie\\"; Minted \\"Chibi Valkyrie\\"; Minted \\"Chibi Valkyrie\\"; Minted \\"Chibi Valkyrie\\""`;

exports[`formatPlain should work as expected > collectible-trade 0x0000020e0e1758d442865233da076d62a5bfa53992229079383096f9fddcc22d 1`] = `"Bought \\"PORSCHÎž 911\\" from 0xdb082B9F5C7748E0fEB4618b8676cB2ddfe89a1E on \\"OpenSea\\""`;

exports[`formatPlain should work as expected > collectible-trade 0x000043e7d363d7d269c279e888cfe6b569e8a48f315736a00bdf06ff8627cafc 1`] = `"Sold \\"(B)APETAVERSE\\" to 0x60f705861aE52E4787543c1c47B6eaf25a4B9ACb on \\"OpenSea\\""`;

exports[`formatPlain should work as expected > collectible-trade 0x6cd99adfdedf1ef1bb7b9958f41425f6ea15a53a7e5dadcac84900d8d179a4d9 1`] = `"Sold \\"Catalog\\" to 0x6170B3C3A54C3d8c854934cBC314eD479b2B29A3 on \\"Zora\\""`;

exports[`formatPlain should work as expected > collectible-trade 0x38baff44e24e7c1df9761f3c9253647ad9edbc08dc254063071b33ff1a4638af 1`] = `"Made a bid for \\"THE MRC ARTE 1/1 COLLECTION\\" on \\"Foundation\\""`;

exports[`formatPlain should work as expected > collectible-trade 0x57b6aa88319e01dc21f01caf898ae04a5bdc7c477da7d562b648ea3563191890 1`] = `"Sold \\"Where the Wild Cats Are\\" to 0xcDA72070E455bb31C7690a170224Ce43623d0B6f on \\"Foundation\\""`;

exports[`formatPlain should work as expected > collectible-trade 0x7076d8902b2256666407108bcaae44a207d37e1b2aac379a87f053aee0af0962 1`] = `"Sold \\"Visage\\" to 0xcDA72070E455bb31C7690a170224Ce43623d0B6f on \\"Foundation\\""`;

exports[`formatPlain should work as expected > collectible-trade 0xb6412dc1f5852f73355f135f2d916b1d37a6d696d305585128c13c2fb1374f3d 1`] = `"Sold \\"Catalog\\" to 0x6170B3C3A54C3d8c854934cBC314eD479b2B29A3 on \\"Zora\\""`;

exports[`formatPlain should work as expected > collectible-transfer 0x000020ba6fd5833d1d54c1be8ee392bf2c5535609d272012a5fe64a23513519c 1`] = `" 0x8aE57A027c63fcA8070D1Bf38622321dE8004c67 Transferred \\"adidas Originals: Into the Metaverse\\" to 0x137ffd2016152a0e90000Dc193fcCe1BE5513E20"`;

exports[`formatPlain should work as expected > donation-donate 0x5a19dd1c81fa00599655bcd4b49ae27baca55ab372a8387d40201bae5ab3eddd 1`] = `"Donated \\"Distributed Solar for Web3 Creators in Nigeria\\" on \\"Gitcoin\\"; Donated \\"Gitcoin Grants Official Matching Pool Fund\\" on \\"Gitcoin\\""`;

exports[`formatPlain should work as expected > exchange-liquidity 0x00003f9c21c7f631474c1bbc12f0946b3f093b520fb707d9cc661a25b58e4ccd 1`] = `"Added 3,569.612 SDAO , 0.8 WETH to liquidity on \\"Uniswap\\""`;

exports[`formatPlain should work as expected > exchange-liquidity 0x0001b57e2fc71a92426ed1bacdd55430767b8c2aa343c777c4daa48de57e8d9e 1`] = `"Removed 37,992.305 TRISM , 0.404 WETH from liquidity on \\"Uniswap\\"; Collected 38,105.289 TRISM , 0.413 WETH from liquidity on \\"Uniswap\\""`;

exports[`formatPlain should work as expected > exchange-liquidity 0x0006b932a5f8aa61e5ac2c62bdc7e8b9b90112418bc22f24604dd421858e6b88 1`] = `"Collected 1,037,965,041.966 SHI , 0.582 WETH from liquidity on \\"Uniswap\\""`;

exports[`formatPlain should work as expected > exchange-liquidity 0x0020cbebcafbbd57dd4a8dbcfca7372f887f25e4662c000cfc9f392362c11912 1`] = `"Supplied 0.25 WETH to liquidity on \\"BendDAO\\""`;

exports[`formatPlain should work as expected > exchange-liquidity 0x0095dd2588fdec8af52a28b792a89f0536745b17bdbace39e392041f1eb7d5a6 1`] = `"Borrowed 2,300.00 DAI from liquidity on \\"AAVE\\""`;

exports[`formatPlain should work as expected > exchange-liquidity 0x008878041c0a55259ce0b2a89eb07a8620b1256006c8edb659baed38bc844b88 1`] = `"Repaid 85,113.76 USDT to liquidity on \\"AAVE\\""`;

exports[`formatPlain should work as expected > exchange-liquidity 0x00c9f3aa1fd06fe17cca96bb444f0aaca610e878626dc9baf489d3550a22041e 1`] = `"WithDrew 1,000,000.00 USDT from liquidity on \\"AAVE\\""`;

exports[`formatPlain should work as expected > exchange-loan 0x0000145456a508beab27e38dbd14fbeec925e95f4f5eda7fc8560f4f5e529820 1`] = `"Refinanced loan 1.203 ETH on \\"Blur\\""`;

exports[`formatPlain should work as expected > exchange-loan 0x07346d4e4c0c2d547ee0927c607a2e1e717507ab01c13175c54d08313eb0c2fb 1`] = `"Repaid loan 40.787 WETH on \\"BendDAO\\""`;

exports[`formatPlain should work as expected > exchange-loan 0x074190b9714fc4e306671252d60245ffb6574e336c174ff1e47f77cebe26891c 1`] = `"Created loan 44.243 WETH on \\"BendDAO\\""`;

exports[`formatPlain should work as expected > exchange-loan 0xa7e278148dbac1685eae105adad72183ed9ea9170a087fb7d2f742b2fd198b1a 1`] = `"Seized loan 0 on \\"Blur\\""`;

exports[`formatPlain should work as expected > exchange-loan 0xbe78726deb3efb4ea062517fe23cdc11312bea4fda8a5903693f9030fd8b472e 1`] = `"liquidated loan 35.463 WETH on \\"BendDAO\\""`;

exports[`formatPlain should work as expected > exchange-swap 0x000005461c52e2eb738744246388f8b04ad94c9ebb4b87668f9f1936b1816512 1`] = `"Swapped 0.000765 ETH to 200.171 JUP on \\"1inch\\""`;

exports[`formatPlain should work as expected > metaverse-mint 0x08effd820f49f2b25c5fcfccedb9384f5d5c35037094b5e492b0c6b75ecfc32b 1`] = `"Minted \\"Mars4\\" on \\"Mars4\\""`;

exports[`formatPlain should work as expected > metaverse-trade 0x001a2d0a33b8e67d8056e94c81e57727541c85906efd0134ff72e0d026a1d47f 1`] = `"Bought \\"Aavegotchi\\" from 0xB415Cc05775Fbcfc68cD80BF58f61D2733614203 on \\"Aavegotchi\\""`;

exports[`formatPlain should work as expected > metaverse-trade 0x03190712b9bf73189595aa26baa69a680856591cf1c51e7949d1604d8ae38e72 1`] = `"Listed \\"FAKE Gotchis\\" from 0xB652158F67b9Fb39c29412d6F8e1C563FF6724f2 on \\"Aavegotchi\\""`;

exports[`formatPlain should work as expected > metaverse-trade 0x2a9b1776be9c0dfad8c17536bbf6b4831b49cf482d186938a3388d094d1175b9 1`] = `"Sold \\"FAKE Gotchis\\" from 0xA7f1C77998bAe58614Be010AD2A806639E280056 on \\"Aavegotchi\\""`;

exports[`formatPlain should work as expected > metaverse-transfer 0x096400ee4246f99aea883ec4bde65c2dd0a23ab708c28bf0540832bf6c69664e 1`] = `"Transferred \\"Aavegotchi FUD\\" to 0x936181C78169f56e2cdd396Fe511a96191a6623E; Transferred \\"Aavegotchi FUD\\" to 0x31b3DB5412fbB08F8C1038f3C6b8DEBa73EF01Ce; Transferred \\"Aavegotchi FOMO\\" to 0x936181C78169f56e2cdd396Fe511a96191a6623E; Transferred \\"Aavegotchi FOMO\\" to 0x31b3DB5412fbB08F8C1038f3C6b8DEBa73EF01Ce; Transferred \\"Aavegotchi ALPHA\\" to 0x936181C78169f56e2cdd396Fe511a96191a6623E; Transferred \\"Aavegotchi ALPHA\\" to 0x31b3DB5412fbB08F8C1038f3C6b8DEBa73EF01Ce; Transferred \\"Aavegotchi KEK\\" to 0x936181C78169f56e2cdd396Fe511a96191a6623E; Transferred \\"Aavegotchi KEK\\" to 0x31b3DB5412fbB08F8C1038f3C6b8DEBa73EF01Ce"`;

exports[`formatPlain should work as expected > social-comment 0x1faacf487d304e601e100b271fcebc108ec45ff7c11fa68962c6cd09576d8eb4 1`] = `" 0xDb46d1Dc155634FbC732f92E853b10B288AD5a1d Commented \\"Beautiful work!\\" on \\"Hey\\""`;

exports[`formatPlain should work as expected > social-delete 0x071c79c1c380c984657b5fe8ea9eb61459e8ac596bbc2a36145a47f7c32c5e3f 1`] = `"Deleted \\"test\\" on \\"Crossbell\\""`;

exports[`formatPlain should work as expected > social-follow 0x6f7f8ee07638f3ff69e901fb4ee348a5f7e37fda99b720d53be0f7b4e1f9e8a4 1`] = `" \\"elshankarimi.lens\\" followed \\"\\" on \\"Lens\\""`;

exports[`formatPlain should work as expected > social-mint 0xafd4f260528c8274345a6447028d84bc2303eddb0e8bca1f2006a106632cda9a 1`] = `"Minted \\"Halfway through week 5 of @verticalcrypto.lens and...\\" on \\"orb\\""`;

exports[`formatPlain should work as expected > social-post 0xa7f1fc265246844407518f165e2d30a70aea79eb14dd0d00deb589b270d7519c 1`] = `"Published post \\"I think they are blended together.\\" on \\"lenstube\\""`;

exports[`formatPlain should work as expected > social-profile 0x01d8db40858063a945cfe5d22b62eeef3371b065f29353a68a0d4f578ca21d1d 1`] = `"Renewed a profile \\"212022.eth\\" on \\"ENS\\""`;

exports[`formatPlain should work as expected > social-profile 0x8e05239a8a48bea6aeec99321a3150a9668945e041ff60e351c640de34cc4b89 1`] = `"Updated a profile \\"0385830.eth\\" on \\"ENS\\""`;

exports[`formatPlain should work as expected > social-profile 0x14717ac164ae9606abb01567d02c9053758b2821a4c666204b730e49e21cd8d1 1`] = `"Created a profile on \\"Lens\\""`;

exports[`formatPlain should work as expected > social-profile 0x5597682570383f1a57a82b3b77673a4561d472d0fdc6ba324d8e687e789c9df9 1`] = `"Wrapped a profile \\"0-770.eth\\" on \\"ENS\\""`;

exports[`formatPlain should work as expected > social-profile 0xe2e6f42795b4bbff284d4d68b68e9099ddb7dcb4dcdbb21add936f0e63e01fa7 1`] = `"Unwrapped a profile \\"notantoine.eth\\" on \\"ENS\\""`;

exports[`formatPlain should work as expected > social-proxy 0x002c594c5c76489420b05ad7f4dda9ee096908290d776a307f0e164980553138 1`] = `"Approved a proxy on \\"Crossbell\\""`;

exports[`formatPlain should work as expected > social-proxy 0x030706597f3c4502896339c8cfd30c088f857abca78033c00cdc3b6605f60c2c 1`] = `"Removed a proxy on \\"Crossbell\\""`;

exports[`formatPlain should work as expected > social-revise GzlgoJetNg9yEI0nRlpx1c4vOww8mDh5Auyy5P9I5uE 1`] = `"Revised \\"Privacy Matters: Newsletter #7\\" on \\"Mirror\\""`;

exports[`formatPlain should work as expected > social-share 0xddd79ba4fde200f82192ca0306097e4c80904e74c96b64684ff66d818b0379a0 1`] = `"Shared \\"death busters stickies ;p\\" on \\"orb\\""`;

exports[`formatPlain should work as expected > social-unfollow 0x27f4849213868ce96c7969c4f652c512ddb659ba5798793ae14c3b24fb55affd 1`] = `" \\"\\" unfollowed \\"flashback.lens\\" on \\"Lens\\""`;

exports[`formatPlain should work as expected > transaction-approval 0x000002b417ce53396ccbe6e2d25f3a41fddae34496241bbbd6bb3bdec1202b84 1`] = `" 0x27c5932eE1B0873e67279066bE914d46203fC738 approved 796.00 APE to 0x5d5092C22c5D0Fd3FEC42b6b2ce796d5B33421C6"`;

exports[`formatPlain should work as expected > transaction-approval 0x00017e026678f05dd015965defe67e336a57afb986a5808cef0a9360e5575659 1`] = `" 0x7dbde20Ce34b0555bcEED3f26B8Bd2653c85045A revoked the approval of 0.00 USDC to 0x57891966931Eb4Bb6FB81430E6cE0A03AAbDe063"`;

exports[`formatPlain should work as expected > transaction-bridge 0x000011998a7ce3715ea30a4264b246175fc1e59621a919e36d3aeb7686c55764 1`] = `"Deposited 0.062 ETH on \\"zkSync Lite\\""`;

exports[`formatPlain should work as expected > transaction-bridge 0x00214bb9008957c1bed2f7d3c9eee8ee4716c4ed29341a9eebc7ac541805929c 1`] = `"Withdrew 0.00550 ETH on \\"zkSync Lite\\"; Withdrew 0.00157 ETH on \\"zkSync Lite\\"; Withdrew 0.00278 ETH on \\"zkSync Lite\\"; Withdrew 0.465 ETH on \\"zkSync Lite\\"; Withdrew 0.001 ETH on \\"zkSync Lite\\"; Withdrew 0.0265 ETH on \\"zkSync Lite\\"; Withdrew 0.00562 ETH on \\"zkSync Lite\\"; Withdrew 0.113 ETH on \\"zkSync Lite\\"; Withdrew 0.465 ETH on \\"zkSync Lite\\"; Withdrew 3.801 ETH on \\"zkSync Lite\\""`;

exports[`formatPlain should work as expected > transaction-burn 0x0012e3add224dcda949c2f04150ee5b719696d2192ecc539c3a4d33f3d4d0821 1`] = `" 0x0063eE34Ed5dc6d0FBAFe66505b9Ad83b23053BE Burned 181.368 HEX"`;

exports[`formatPlain should work as expected > transaction-deploy 0x01d8ac59932c90ea081feca9c5482b0da5881ea69044ab845994865325687234 1`] = `"Deployed a contract 0xF88FD34aD1f4B6eF690a8abAb9182b001199D85D"`;

exports[`formatPlain should work as expected > transaction-mint 0x00003fc6bea1d10b9110d631c8eeb84f2bb4450389b774d53dda9dd6743561ab 1`] = `" 0x0000000000000000000000000000000000000000 Minted 27,717,258.00 XEN"`;

exports[`formatPlain should work as expected > transaction-multisig 0x00005c195a19697b221ea6341cdd61f439e37d7193391a96d114c2df83300c90 1`] = `"Created a multisig transaction to 0xa6B71E26C5e0845f74c812102Ca7114b6a896AB2 on \\"Safe\\""`;

exports[`formatPlain should work as expected > transaction-multisig 0x002fc61cef05d5f6e6c54b236dc7c817233950065264e73e979b435e9d83a6d6 1`] = `"Added 0x34849637366646E941069B84df256E5F45597944 to 0x59d9AB9C68ef7357de1B96DE91ce7c8f14BB8FE8 on \\"Safe\\"; Executed a multisig transaction on \\"Safe\\""`;

exports[`formatPlain should work as expected > transaction-multisig 0x004be0e60b8375334b14caf6ac0f6beec0de419f841b332746a4a2f77e1a5d1c 1`] = `"Executed a multisig transaction on \\"Safe\\""`;

exports[`formatPlain should work as expected > transaction-multisig 0x0013bc2054920660ca961c5cba695cc87a6845dc33d359dbe227865d3803dde9 1`] = `"Changed the threshold of 0xBcCec010d81Fd2001D063F0a5a8c9f7f69a87396 on \\"Safe\\"; Executed a multisig transaction on \\"Safe\\""`;

exports[`formatPlain should work as expected > transaction-multisig 0x0036591c0ac8c2b0d06bc7080e1c6851190366b917dc0101f9026b667e58bdbc 1`] = `"Removed 0xfD523eED91f509C10800DaCcBffD20270385bbCB from 0xB49EB4543f7B82D2e8fA114C02567A5E1ED4E6c9 on \\"Safe\\"; Added 0x0890aC039CE5D07928576c28CF500E82AD45E3f9 to 0xB49EB4543f7B82D2e8fA114C02567A5E1ED4E6c9 on \\"Safe\\"; Executed a multisig transaction on \\"Safe\\""`;

exports[`formatPlain should work as expected > transaction-transfer 0x000000c2fb1a34a175aa4bb001ac3b62c2fe62364e7cbe62aab9f635391f9fd1 1`] = `" 0x5c5Fb3359461bBA32BFf65eE1e1a9A11F8279c9B sent 0.980 ETH to 0xA9D1e08C7793af67e9d92fe308d5697FB81d3E43"`;
